http.go

```diff
@@ -322,7 +329,87 @@ func httpPutViolation(w http.ResponseWriter, r *http.Request) {
 	// Force object field and type to match value specified in request path
 	v.Object = valstr
 	v.Type = typestr
-	httpPutViolationsInner(w, r, typestr, []ViolationRequest{v})
+	if err := putViolation(typestr, v); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	return
+}
+
+func putViolation(typestr string, v ViolationRequest) error {
+	v.Fixup(typestr)
+	// Force type field to match value specified in request path
+	v.Type = typestr
+	err := v.Validate()
+	if err != nil {
+		return err
```

Comment:
With the error logging gone here are we also going to lose logging errors that result from the call to `httpPutViolations`?